### DISCUSSION TITLE
WYSIWYG: Link to The File from library

### DISCUSSION THREAD

**Discussion by AndreyKindin** [directus/directus#14672] [Feature Requests]
In WYSIWYG it is currently not possible to create a link to a file from the library.

When clicking on link(chain) icon, there is no possibility to create a link to a file from the library

**Comment by rijkvanzanten** [comment]
I'm confused, are you trying to generate a link to the /assets endpoint?

**Reply by AndreyKindin** [reply]
Yes.
Many of our customers need it to e.g.: link pdfs directly in the text

Example:

https://user-images.githubusercontent.com/9415025/181037239-50bca2a0-8cbc-4d30-b579-3f450cba8b4f.mp4

**Reply by rijkvanzanten** [reply]
How would you handle file permissions in that case? When doing this, we would also have to add a way to add a (static?) token just like we do with the regular images, and potentially a way to add the `download` attribute on the `<a>` tag it generates 🤔

**Reply by AndreyKindin** [reply]
In my concrete case all files are publicly readable, this is probably not the standard, but the access rights can be set via the roles & permissions.
As for download it is easier to attach ?download then the file is also downloaded mandatory

**Reply by rijkvanzanten** [reply]
> As for download it is easier to attach ?download then the file is also downloaded mandatory

Yeah, though I don't know what repercussions that has on the a11y of the `<a>` tag. I'm assuming the `download` attribute on anchors gives browsers and a11y software other hints on the link type, but I'll have to research that more

**Reply by AndreyKindin** [reply]
I can add both, it will not hurt

**Comment by nervusdm** [comment]
This will be a great feature

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
